Hardware/Hardware/CardReader/StripeCardReader/PaymentIntent+Stripe.swift

```diff
@@ -10,7 +10,7 @@ extension PaymentIntent {
         self.created = intent.created
         self.amount = intent.amount
         self.currency = intent.currency
-        self.metadata = intent.metadata
+        self.metadata = intent.metadata as? [String: String]
```

Comment:
I didn't realize there was a `metadata` type coming from StripeTerminal already that's defined as `[AnyHashable: Any]`. I still think it's reasonable to expect it to only contain strings, but I'm a bit more worried than before, since I think this cast would fail for the whole dictionary if a single key or value isn't a string?

Reply:
Yeah, agreed, and we're relying on the Stripe SDK maintaining that behaviour. Consequences of that seem fairly small at the moment, but having worked on another receipt field I might come back to this and rework it. For now I think it's fine, so I'm going to merge.